inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/detection_output/detection_output_kernel_ref.cpp

```diff
@@ -0,0 +1,249 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "detection_output_kernel_ref.h"
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+ParamsKey DetectionOutputKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableInputLayout(DataLayout::bfyx);
+    k.EnableOutputLayout(DataLayout::bfyx);
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    return k;
+}
+
+JitConstants DetectionOutputKernelRef::GetJitConstants(const detection_output_params& params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+
+    const auto& detectOutParams = params.detectOutParams;
+
+    jit.AddConstants({
+        MakeJitConstant("NUM_IMAGES", detectOutParams.num_images),
+        MakeJitConstant("NUM_CLASSES", detectOutParams.num_classes),
+        MakeJitConstant("NUM_CLASSES_PER_ITEM", 4),
+        MakeJitConstant("KEEP_TOP_K", detectOutParams.keep_top_k),
+        MakeJitConstant("TOP_K", detectOutParams.top_k),
```

Comment:
how about set the TOP_K as effective number, min(detectOutParams.top_k, # of priors) ?

Reply:
Applied as your suggestion